libsolidity/formal/CHC.h

```diff
@@ -207,7 +207,7 @@ class CHC: public SMTEncoder
 	//@}
 
 	/// CHC solver.
-	std::shared_ptr<smt::CHCSolverInterface> m_interface;
+	std::unique_ptr<smt::CHCSolverInterface> m_interface;
```

Comment:
Here one could also think about moving the declaration up... but yeah - aparently nothing really depends on it and destruction order doesn't really matter in the end, so actually no need...